test/node-interaction.js

```diff
@@ -0,0 +1,317 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Enables ES2015 import/export in Node.js
+
+const {execFile} = require('child_process');
+const puppeteer = require('puppeteer');
+const console = require('console');
+const process = require('process');
+
+const fs = require('fs');
+const path = require('path');
+const PNG = require('pngjs').PNG;
+const pixelmatch = require('pixelmatch');
+
+const LIB_DIR = path.resolve(__dirname, './interaction');
+
+function printResult(diffRatio, threshold) {
+  return diffRatio <= threshold
+    ? console.log('\x1b[32m%s\x1b[0m', 'Rendering test Passed!')
+    : console.log('\x1b[31m%s\x1b[0m', 'Rendering test failed!');
+}
+
+const innerWidth = 1000;
+const innerHeight = 800;
+
+const EVENTS = ['', '-pan-ur', '-pan-dl', '-tilt-ur', '-tilt-dl', '-zoom-in', '-zoom-out'];
+
+const DIRECTION = {
+  UP: 0,
+  RIGHT: 1,
+  DOWN: 2,
+  LEFT: 3,
+  UPRIGHT: 4,
+  DOWNLEFT: 5
+};
+
+async function pan(page, distance, direction) {
+  const mouse = page.mouse;
+  const centerx = innerWidth / 2;
+  const centery = innerHeight / 2;
+  await mouse.move(centerx, centery);
+  await mouse.down();
+  await page.waitFor(2000);
+  // when mouse.move and mouse.down events are triggered by Puppeteer,
+  // the starting position is off by several pixels, 2 seconds waiting time
+  // is needed remove it.
+  switch (direction) {
+    case DIRECTION.UP:
+      await mouse.move(centerx, centery - distance, {steps: distance});
+      break;
+    case DIRECTION.RIGHT:
+      await mouse.move(centerx + distance, centery, {steps: distance});
+      break;
+    case DIRECTION.DOWN:
+      await mouse.move(centerx, centery + distance, {steps: distance});
+      break;
+    case DIRECTION.LEFT:
+      await mouse.move(centerx - distance, centery, {steps: distance});
+      break;
+    case DIRECTION.UPRIGHT:
+      await mouse.move(centerx + distance, centery - distance, {steps: distance});
+      break;
+    case DIRECTION.DOWNLEFT:
+      await mouse.move(centerx - distance, centery + distance, {steps: distance});
+      break;
+    default:
+      console.log('Please input a number between 0 and 5!');
+      break;
+  }
+  await mouse.up();
+}
+
+async function panAll(page, distance, waitingTime, threshold, folder) {
+  await pan(page, distance, DIRECTION.UPRIGHT);
+  await page.screenshot({path: `${folder}${EVENTS[1]}.png`});
+  await pan(page, distance, DIRECTION.DOWNLEFT);
+  await pan(page, distance, DIRECTION.DOWNLEFT);
+  await page.screenshot({path: `${folder}${EVENTS[2]}.png`});
+  await pan(page, distance, DIRECTION.UPRIGHT);
+}
+
+async function tilt(page, angle, direction) {
+  const mouse = page.mouse;
+  const centerx = innerWidth / 2;
+  const centery = innerHeight / 2;
+  await mouse.move(centerx, centery);
+  await mouse.down();
+  await page.waitFor(2000);
```

Comment:
Make 2000 a constant that is shared cross all functions.

Reply:
@Pessimistress updated